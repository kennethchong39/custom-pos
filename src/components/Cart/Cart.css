.current-order {
  overflow: scroll;
  padding: 0 10px;
}

.current-order-header {
  padding-top: 10px;
  text-align: center;
  position: sticky;
  top: 0;
  background-color: white;
}

.current-order-header h2 {
  margin: 0px;
}

.current-orders {
  display: flex;
  flex-direction: column;
  overflow: hidden;
  margin: 0px 15px;
}

.current-item {
  display: flex;
  font-size: 1.5rem;
  padding: 0 15px;
  justify-content: space-between;
  border-bottom: 1px solid grey;
}

.current-item-empty {
  text-align: center;
  margin: 3.5rem;
  font-size: 1.3rem;
  color: grey;
}

.current-order-footer {
  display: flex;
  flex-direction: column;
  font-size: 1.5rem;
  align-items: center;
  margin-bottom: 30px;
}

.current-order-total {
  display: flex;
  justify-content: space-between;
  padding: 0 15px;
  margin: 0 10px 10px;
  font-size: 1.5rem;
  font-weight: 800;
  width: 95%;
  border-top: 1px solid black;
}

.current-order-footer button {
  border: none;
  padding: 15px 5px;
  font-size: 1.3rem;
  width: 95%;
  background-color: #ff3a33;
  color: white;
}

.disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.current-item:last-child {
  border: none;
}
