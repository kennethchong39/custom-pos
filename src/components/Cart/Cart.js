import React, { useState, useEffect } from 'react';

import './Cart.css';

export default function Cart(props) {
  // to keep track of the total amount
  const [total, setTotal] = useState(0);

  // to update the price when an order is added to the cart
  useEffect(() => {
    let sum = 0;
    for (const items of props.items) {
      sum += items.price * items.quantity;
    }
    setTotal(sum.toFixed(2));
  }, [props.items]);

  // submit the order once the submit button is clicked
  const handleSubmit = () => {
    props.handleSubmit(total);
    setTotal(0);
  };

  const totalItems = props.items.length;

  return (
    <div className="current-order">
      <div className="current-order-header">
        <h2>Cart</h2>
        <hr />
      </div>
      <div className="current-orders">
        {totalItems === 0 && (
          <div className="current-item-empty">
            <p>Empty Cart</p>
          </div>
        )}
        {props.items.map((item, index) => {
          return (
            <div className="current-item" key={index}>
              <p>{item.name}</p>
              <p>x {item.quantity}</p>
              <p>$ {item.price}</p>
            </div>
          );
        })}
      </div>
      <div className="current-order-footer">
        <div className="current-order-total">
          <p>Total</p>
          <p>$ {total}</p>
        </div>
        <button
          disabled={totalItems === 0 ? true : false}
          className={totalItems === 0 ? 'disabled' : ''}
          onClick={handleSubmit}
        >
          Submit Order
        </button>
      </div>
    </div>
  );
}
