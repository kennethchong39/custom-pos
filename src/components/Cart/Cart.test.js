import renderer from 'react-test-renderer';

import Cart from './Cart';

test('component: Cart tests', () => {
  const testProps = [
    {
      name: 'Vegetarian',
      price: 8.99,
      ingredients: {
        bread: 2,
        lettuce: 2,
        tomato: 2,
        cheese: 2,
      },
      quantity: 1,
    },
  ];
  const tree = renderer.create(<Cart items={testProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
