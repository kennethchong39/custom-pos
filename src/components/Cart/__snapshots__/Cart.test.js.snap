// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Cart tests 1`] = `
<div
  className="current-order"
>
  <div
    className="current-order-header"
  >
    <h2>
      Cart
    </h2>
    <hr />
  </div>
  <div
    className="current-orders"
  >
    <div
      className="current-item"
    >
      <p>
        Vegetarian
      </p>
      <p>
        x 
        1
      </p>
      <p>
        $ 
        8.99
      </p>
    </div>
  </div>
  <div
    className="current-order-footer"
  >
    <div
      className="current-order-total"
    >
      <p>
        Total
      </p>
      <p>
        $ 
        0
      </p>
    </div>
    <button
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Submit Order
    </button>
  </div>
</div>
`;
